### DISCUSSION TITLE
Automatically Registering a New Channel if It Doesn't Exist

### DISCUSSION THREAD

**Discussion by rfscheidt** [bluenviron/mediamtx#3786] [General]
Hello everyone,

Is there a way to automatically register a new channel if it doesn't already exist? For example, when attempting to access the following RTSP URL:

rtsp://18.18.18.18:8554/channel1?user=x&pass=y&ip=18.18.18.18&port=8554&chan=Streaming/Channels/301

Thanks in advance for your help!

**Comment by rfscheidt** [comment]
Hi @aler9 , sorry for insisting, but I would like to try creating a feature to do this.
Could you please just give me some guidance?

I would like to create a new path when receiving an RTSP call.
What would be the best place to add this part of the code?

I tried creating it in the method func (m *Manager) authenticateHTTP(req *Request) error { before authentication. But it doesn't work; it seems to restart the application's context and cancels the RTSP call.

func (m *Server) createChannelHTTP() error {
confName := "namechanneltest"

pathJson := fmt.Sprintf({ "name": "%s", "source" : "rtsp://xxxx:xxx@127.0.0.1:8554/%s" }, confName, confName)

var p conf.OptionalPath
err := json.Unmarshal([]byte(pathJson), &p.Values)
if err != nil {
fmt.Println("Error parsing JSON:", err)
return nil
}

//a.mutex.Lock()
//defer a.mutex.Unlock()

newConf := m.Conf.Clone()

err = newConf.AddPath(confName, &p)
if err != nil {
return nil
}

err = newConf.Validate()
if err != nil {
return nil
}

m.Conf = newConf
//m.Parent.APIConfigSet(newConf)

return nil
}

**Reply by aler9** [reply]
Hello, i perfectly understand the usefulness of this feature (since it allows to tune path settings independently for each path) but it's something that is currently unimplemented. You can either create a path configuration before starting the RTSP connection (for instance, through an API call) or you can use regex path configurations.
The feature you want is being tracked here: #3718
Regarding implementation, it's not trivial but not impossible too, and currently i can't help you with it. Therefore you have to evaluate whether it's better to go on developing or just switch to the API+RTSP method.

[a comment by an automated account is not shown]
